ctapipe/io/astropy_helpers.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import tables
+from astropy.table import Table
+from astropy.units import Unit
+
+__all__ = ["h5_table_to_astropy"]
+
+
+def h5_table_to_astropy(h5file, path) -> Table:
+    """Get a table from a ctapipe-format HDF5 table as an `astropy.table.Table`
+    object, retaining units. This uses the same unit storage convention as
+    defined by the `HDF5TableWriter`, namely that the units are in attributes
+    named by `<column name>_UNIT` that are parsible by `astropy.units`.
+
+    Parameters
+    ----------
+    h5file: Union[str, tables.file.File]
+       input filename or PyTables file handle
+    path: str
+       path to table in the file
+
+    """
+
+    should_close_file = False
+    if not isinstance(h5file, tables.file.File) and isinstance(h5file, str):
+        h5file = tables.open_file(h5file)
+        should_close_file = True
+    else:
+        raise ValueError(
+            f"expected a string or PyTables filehandle for argument 'h5file', got {h5file}"
+        )
+
+    table: tables.table.Table = h5file.get_node(path)
+
+    column_units = {}  # mapping of colname to unit
+    for attr in table.attrs._f_list():
```

Comment:
I would maybe check the other way around. So go over all columns and check if the expected field is there.

This would also not rely on private parts:

```
units = {col: table.attrs.get('f{col}_UNIT'} for col in table.dtype.names}
```
and then pass this to the `Table` constructor

Reply:
Sorry, it has no get method

Reply:
> also not rely on private parts

those are not really private, see the pytables docs. They reserve `_f_` as public functions to avoid conflicting with the fact that you can access columns and attributes by `x.attr_name`